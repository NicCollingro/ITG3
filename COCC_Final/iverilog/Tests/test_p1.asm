.text

alutest:
	ldi A 0xFF
	ldi B 0x0F
	and
	mov G A
	rout A

	ldi A 0xFF
	or
	mov F A
	stx A %out
	mout %out		; OUT: 0xff

	ldi A 0xFF
	xor
	mov E A	
	stx A %out
	mout %out		; OUT: 0xf0

	ldi A 0xFF
	add
	mov D A
	stx A %out
	mout %out		; OUT: 0x(1)0e 

	ldi A 1
	ldi B 2
	sub
	stx A %out
	mout %out		; OUT: 0x(1)ff
	jmp %calltest

calltest:
	ldi A 0x0A
	stx A %out
	mout %out		
	call %load
	stx A %out
	mout %out		; OUT: 0x0a, 0x2a, 0x0a
	jmp %multest		
load:
	push A
	ldi A 42
	call %done
	pop A
	ret
done:
	stx A %out
	mout %out
	ret

multest:
	ldx A %r
	ldi B 4
	add
	stx A %r

	ldx A %i
	dec
	stx A %i

	jnz %multest

	ldx A %r
	stx A %out
	mout %out		; OUT: 0x14
	jmp %divtest

divtest:
	ldx A %n	
	ldx B %d	
	ldi C 0		
loopdivtest:
	cmp 
	jc %donedivtest	
	sub		
	stx A %n	
	mov A C		
	inc		
	mov C A		
	ldx A %n	
	jmp %loopdivtest	
donedivtest:
	stx A %out
	mout %out		; OUT: 0x04
	mov B A		
	mov A C		
	stx A %out
	mout %out		; OUT: 0x19
	jmp %alumultest

alumultest:
	ldx B %fb
	ldx A %fa
	mlo
	stx A %out
	mout %out		; OUT: 0xf4
	ldx A %fa
	mhi
	stx A %out
	mout %out		; OUT: 0x03
	jmp %nottest

nottest:
	ldi A 0xaa
	not
	stx A %out
	mout %out		; OUT: 0x55
	jmp %shifttest

shifttest:
	ldi A 0xaa
	ldi B 4
	shl
	stx A %out
	mout %out		; OUT: 0xa0
	rol
	stx A %out
	mout %out		; OUT: 0x41
	ldi B 2
	shr
	stx A %out
	mout %out		; OUT: 0x10
	jmp %sqrttest

sqrttest:
	ldi A 128		
	sqrt
	stx A %out
	mout %out		; OUT: 0x0b
	ldi A 255		
	sqrt
	stx A %out
	mout %out		; OUT: 0x0f
	hlt

.data
	n = 129		; ZÃ¤hler
	d = 5		; Nenner
	r = 8
	i = 3
	s = 4
	fa = 44
	fb = 23
	out = 0x5a
