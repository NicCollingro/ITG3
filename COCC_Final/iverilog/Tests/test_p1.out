    2680 OUTPUT:  15 (0x0f 0b00001111)
    5800 OUTPUT: 255 (0xff 0b11111111)
    8920 OUTPUT: 240 (0xf0 0b11110000)
   12040 OUTPUT:  14 (0x0e 0b00001110)
   15280 OUTPUT: 255 (0xff 0b11111111)
   17920 OUTPUT:  10 (0x0a 0b00001010)
   22480 OUTPUT:  42 (0x2a 0b00101010)
   26080 OUTPUT:  10 (0x0a 0b00001010)
   44680 OUTPUT:  20 (0x14 0b00010100)
  181840 OUTPUT:   4 (0x04 0b00000100)
  184240 OUTPUT:  25 (0x19 0b00011001)
  188320 OUTPUT: 244 (0xf4 0b11110100)
  191080 OUTPUT:   3 (0x03 0b00000011)
  194320 OUTPUT:  85 (0x55 0b01010101)
  198160 OUTPUT: 160 (0xa0 0b10100000)
  200200 OUTPUT:  65 (0x41 0b01000001)
  202840 OUTPUT:  16 (0x10 0b00010000)
  206080 OUTPUT:  11 (0x0b 0b00001011)
  208720 OUTPUT:  15 (0x0f 0b00001111)
REGISTERS: A: 0f B: 02 C: 19 D: 0e E: f0 F: ff G: 0f H: 3f
FLAGS:     CF: 0 ZF: 0
